src/store/index.ts

```diff
@@ -4,57 +4,19 @@ import { createDirectStore } from 'direct-vuex';
 import { Simulation } from 'd3-force';
 
 import {
-  Link, Node, Network, SimulationLink,
+  Link, Node, Network, SimulationLink, State, LinkStyleVariables, LoadError, NestedVariables, ProvenanceEventTypes,
 } from '@/types';
 import api from '@/api';
 import { GraphSpec, RowsSpec, TableRow } from 'multinet';
 import {
-  scaleLinear, ScaleLinear, scaleOrdinal, ScaleOrdinal,
+  scaleLinear, scaleOrdinal,
 } from 'd3-scale';
 import { schemeCategory10 } from 'd3-scale-chromatic';
+import { initProvenance } from '@visdesignlab/trrack';
+import { updateProvenanceState } from '@/lib/provenanceUtils';
 
 Vue.use(Vuex);
 
-interface LoadError {
-  message: string;
-  buttonText: string;
-  href: string;
-}
-
-interface NestedVariables {
-  bar: string[];
-  glyph: string[];
-}
-
-interface LinkStyleVariables {
-  width: string;
-  color: string;
-}
-
-interface AttributeRanges {
-  [key: string]: {attr: string; min: number; max: number};
-}
-
-export interface State {
-  workspaceName: string | null;
-  networkName: string | null;
-  network: Network | null;
-  selectedNodes: Set<string>;
-  loadError: LoadError;
-  simulation: Simulation<Node, SimulationLink> | null;
-  renderNested: boolean;
-  markerSize: number;
-  fontSize: number;
-  labelVariable: string;
-  colorVariable: string;
-  selectNeighbors: boolean;
-  nestedVariables: NestedVariables;
-  linkVariables: LinkStyleVariables;
-  attributeRanges: AttributeRanges;
-  nodeColorScale: ScaleOrdinal<string, string>;
-  linkWidthScale: ScaleLinear<number, number>;
-}
```

Comment:
Why move this stuff out? They're types intimately connected to state tracking, so aren't really generally consumable types outside of the project.

Not saying you can't move these to a top level types.ts file, but I was wondering about your reasoning.

Reply:
Having the types here creates a cyclic import between the provenance library and the vuex store. We need the types in the provenance and we need the provenance in the store. I figured the easiest resolution was to just move the types to types, but if you have some other ideas, I'm open to them

Reply:
Oh ok. Yeah cyclic dependencies are a pain. You're right, the best (only?) thing to do is to factor out the cyclically dependent pieces into a common location.